pkg-manager/plugin-commands-installation/src/recursive.ts

```diff
@@ -291,8 +292,8 @@ export async function recursive (
       const promises: Array<Promise<void>> = mutatedPkgs.map(async ({ originalManifest, manifest, rootDir }) => {
         return manifestsByPath[rootDir].writeProjectManifest(originalManifest ?? manifest)
       })
-      if (updatedCatalogs) {
-        promises.push(addCatalogs(opts.workspaceDir, updatedCatalogs))
+      if (updatedCatalogs ?? opts.cleanupUnusedCatalogs) {
+        promises.push(updateWorkspaceManifest(opts.workspaceDir, { updatedCatalogs, cleanupUnusedCatalogs: opts.cleanupUnusedCatalogs }))
```

Comment:
I see below updateWorkspaceManifest is called again. Let's make sure it isn't called twice.

Reply:
It should not be executed twice. True is returned in the following lines, and the function is exited early.